resource_calendar_rrule/models/resource_calendar.py

```diff
@@ -0,0 +1,250 @@
+                            self.env['resource.calendar.attendance']
+                            ._fields['dayofweek'].selection[day['day']][1],
+                            'rrule': [day_rule] + exdate + rdate,
+                            'hour_from': morning_hour,
+                            'hour_to': morning_hour + day['morning'],
+                        })
+                    )
+                if day['afternoon']:
+                    result.append(
+                        (0, 0, {
+                            'name': _('Simplified attendance %s afternoon') %
+                            self.env['resource.calendar.attendance']
+                            ._fields['dayofweek'].selection[day['day']][1],
+                            'rrule': [day_rule] + exdate + rdate,
+                            'hour_from': afternoon_hour,
+                            'hour_to': afternoon_hour + day['afternoon'],
+                        })
+                    )
+        return result
+
+    @api.multi
+    def get_attendances_for_weekday(self, day_dt):
+        return sum((
+            self.env['resource.calendar.attendance'].new({
+                'name': attendance.name,
+                'dayofweek': str(day_dt.weekday()),
+                'hour_from': attendance.hour_from,
+                'hour_to': attendance.hour_to,
+                'date_from': attendance.date_from,
+                'date_to': attendance.date_to,
+            })
+            for attendance in self.mapped('attendance_ids')
+            if (
+                not attendance.date_from or
+                fields.Date.from_string(attendance.date_from) <=
+                day_dt.date()
+            ) and (
+                not attendance.date_to or
+                fields.Date.from_string(attendance.date_to) >=
+                day_dt.date()
+            ) and any(
+                date1.weekday() == day_dt.weekday() or
+                date2.weekday() == day_dt.weekday()
+                for date1, date2 in attendance._iter_rrule(
+                    *self._get_check_interval(
+                        fields.Datetime.from_string(attendance.date_from),
+                        fields.Datetime.from_string(
+                            attendance.date_to and
+                            attendance.date_to + ' 23:59:59' or None
+                        )
+                    )
+                )
+            )
+        ), self.env['resource.calendar.attendance'])
+
+    @api.multi
+    def get_weekdays(self, default_weekdays=None):
+        if not self:
+            return super(ResourceCalendar, self).get_weekdays(
+                default_weekdays=default_weekdays
+            )
+        return list(set(
+            date.weekday()
+            for attendance in self.mapped('attendance_ids')
+            for date, _ in attendance._iter_rrule(
+                *self._get_check_interval()
+            )
+        ))
```

Comment:
I currently can't find a good reason why we didn't do (3) in the first place. Maybe that was a well meant but misguided optimization.
Concerning optimization, I think we should try to create more efficient rrules, especially when using the simplified calendar. I noticed quite a speed penalty since we introduced this. So for example if a person works full time, it should not generate 10 rules, one for every morning, one for every afternoon, but just one for all afternoons and one for all mornings. Or (and I think that's not as crazy as it sounds) just precompute asynchronously all working intervals for [-N, +M] years, and replace all of the expensive computations by an SQL lookup. And if a request comes in for a not yet computed interval, it's computed on the fly.

Reply:
Hm, (3) in practice (simply returning all days from get_weekdays) interferes with other uses of get_weekdays eg in [get_previous_day](https://github.com/OCA/OCB/blob/10.0/addons/resource/models/resource.py#L215). Went for 2 in the end.

Reply:
https://github.com/sunflowerit/hr/commit/782e18083c22d2dc14d0c33fda3f026c3ecf4fb6